File: app/src/main/java/dmcs/astroWeather/task/DownloadWoeidByLatitudeAndLongitudeTask.java
package dmcs.astroWeather.task;

import android.os.AsyncTask;

import org.json.JSONException;

import java.io.IOException;

import dmcs.astroWeather.util.WeatherDownloader;

/**
 * @Author Mateusz Wieczorek on 2016-07-12.
 */
public class DownloadWoeidByLatitudeAndLongitudeTask extends AsyncTask<String, String, String> {

    @Override
    protected String doInBackground(String... params) {
        try {
            return WeatherDownloader.getWoeidByLatitudeAndLongitude(params[0], params[1]);
        } catch (IOException | JSONException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: app/src/main/java/com/weather/task/DownloadWoeidByLatitudeAndLongitudeTask.kt
package com.weather.task

import android.os.AsyncTask

import com.weather.util.WeatherDownloader

import org.json.JSONException

import java.io.IOException


/**
 * @author Mateusz Wieczorek
 */
class DownloadWoeidByLatitudeAndLongitudeTask : AsyncTask<String, String, String>() {

    override fun doInBackground(vararg params: String): String? {
        try {
            return WeatherDownloader.getWoeidByLatitudeAndLongitude(params[0], params[1])
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return null
    }
}
